extensions/amp-accordion/1.0/accordion.js

```diff
@@ -299,22 +310,25 @@ export function AccordionSection({
     }
   }, [registerSection, id, defaultExpanded]);
 
-  const expandHandler = useCallback(() => {
-    if (toggleExpanded) {
-      toggleExpanded(id);
-    } else {
-      setExpandedState((prev) => {
-        const newValue = !prev;
-        Promise.resolve().then(() => {
-          const onExpandStateChange = onExpandStateChangeRef.current;
-          if (onExpandStateChange) {
-            onExpandStateChange(newValue);
-          }
+  const toggleHandler = useCallback(
+    (opt_expand) => {
+      if (toggleExpanded) {
+        toggleExpanded(id, opt_expand);
+      } else {
+        setExpandedState((prev) => {
+          const newValue = opt_expand || !prev;
+          Promise.resolve().then(() => {
```

Comment:
Sure, all of the code that fires events (basically any time that `onExpandStateChange` is called) is done inside a `Promise.resolve()` so that it can execute after the current render.  W/o explicitly forcing the event to run after the current thread, there could be a lot of timing issues.

For example - The accordion can support toggling all sections at once.  When a particular section is toggled it can send an event to say collpase (or expand) another section.  That other section is also going to be toggled by the original toggling all sections.  So which should resolve first?

So this way we explicitly say, all toggling happens.  THEN all events are triggered and further collapsing / expanding happens after the initial toggling.

Reply:
To clarify, the async even call here is necessary because we're using `setState(function() {})` form of the set state. Unfortunately, that form (itself asynchronous) doesn't "like" even calls from it. But I think the promise here is ok for now. We'll likely come up with some nice `useAsyncCallback` hook eventually for this. This is rare, but we had a couple of cases where this happened already.